### PR TITLE
Add alternative solutions for Euler project problems

### PR DESCRIPTION
**Pull request by gkarwchan**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: project_euler/problem_01/sol8.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+"""
+Problem Statement:
+If we list all the natural numbers below 10 that are multiples of 3 or 5,
+we get 3,5,6 and 9. The sum of these multiples is 23.
+Find the sum of all the multiples of 3 or 5 below N.
+"""
+
+
+def solution(n):
```
Comment (onlinejudge95) [inline project_euler/problem_01/sol8.py:9]: Needs typehints

File: project_euler/problem_02/sol6.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    1,2,3,5,8,13,21,34,55,89,..
+
+By considering the terms in the Fibonacci sequence whose values do not exceed
+n, find the sum of the even-valued terms. e.g. for n=10, we have {2,8}, sum is
+10.
+"""
+
+
+def solution(n):
```
Comment (onlinejudge95) [inline project_euler/problem_02/sol6.py:14]: Needs type hints

File: project_euler/problem_03/sol04.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        ...
+    TypeError: Parameter n must be int or passive of cast to int.
+    >>> solution("asd")
+    Traceback (most recent call last):
+        ...
+    TypeError: Parameter n must be int or passive of cast to int.
+    """
+    try:
+        n = int(n)
+    except (TypeError, ValueError):
```
Comment (onlinejudge95) [inline project_euler/problem_03/sol04.py:45]: Only `ValueError` suffices, I can't see of an example where `int()` raises a `TypeError`

File: project_euler/problem_03/sol04.py [hunk 4]
```diff
@@ -31,21 +31,12 @@ def solution(n):
...
         ...
     ValueError: Parameter n must be greater or equal to one.
-    >>> solution([])
-    Traceback (most recent call last):
-        ...
-    TypeError: Parameter n must be int or passive of cast to int.
-    >>> solution("asd")
-    Traceback (most recent call last):
-        ...
-    TypeError: Parameter n must be int or passive of cast to int.
```
Comment (cclauss) [inline project_euler/problem_03/sol04.py]: Why remove these test?

File: project_euler/problem_03/sol04.py [hunk 5]
```diff
@@ -31,21 +31,12 @@ def solution(n):
...
     """
-    try:
-        n = int(n)
-    except (TypeError, ValueError):
-        raise TypeError("Parameter n must be int or passive of cast to int.")
```
Comment (cclauss) [inline project_euler/problem_03/sol04.py]: Why remove these check?  We can specify types but callers are free to ignore them.

Comment (onlinejudge95) [reply project_euler/problem_03/sol04.py]: Exactly, my change request was to remove `TypeError` **only**

File: project_euler/problem_01/sol8.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    >>> solution(3)
+    0
+    >>> solution(4)
+    3
+    >>> solution(10)
+    23
+    >>> solution(600)
+    83700
+    """
+    return sum(set(range(0, n, 3)).union(set(range(0, n, 5))))
```
Comment (cclauss) [inline project_euler/problem_01/sol8.py:27]:
```suggestion
    return sum(set(range(0, n, 3)) | set(range(0, n, 5)))
```

Comment (onlinejudge95) [reply project_euler/problem_01/sol8.py:27]: Just an opinion I find using the method in comparison to the operator versions seems more pythonic

File: project_euler/problem_01/sol8.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    It uses Python features to do some logic,
+    like using range's step to check for only multiples of 3 and 5
+    and use set's union to remove duplication
+
+    >>> solution(3)
+    0
+    >>> solution(4)
+    3
+    >>> solution(10)
+    23
```
Comment (cclauss) [inline project_euler/problem_01/sol8.py:23]:
```suggestion
    >>> [solution(i) for i in range(11)]
    [ ... ]
```

Comment (cclauss) [reply project_euler/problem_01/sol8.py:23]: Also what happens when we `solution(-1)` or `solution(3.3)`

Comment (onlinejudge95) [reply project_euler/problem_01/sol8.py:23]: The problem statement has a constraint of natural numbers less than 100

Comment (cclauss) [reply project_euler/problem_01/sol8.py:23]: Then the code needs to raise ValueError n is not a natural number < 100.

File: project_euler/problem_03/sol04.py (hunk 4 above)
Comment (gkarwchan) [reply project_euler/problem_03/sol04.py]: @onlinejudge95 told  me to remove the TypeError exception, so we only pass integer and those tests become irrelevant as there is no need to check for string.

Anyway, I thought the main idea here is the Algorithm itself, and not getting busy with all validation.

**Comment by gkarwchan** [comment]
I didn't know that we have to spend lots of code on adding all validations.
Actually I copied the tests/exceptions from previous solution on the same folder, and that code was checked in without problems.
I thought the idea is to present a new algorightm to solve a problem, but not to add all validations of user input.
We could go more and add more and more validations, because user input could be anything.
I am closing this PR

File: project_euler/problem_03/sol04.py (hunk 3 above)
Comment (cclauss) [reply project_euler/problem_03/sol04.py]: `>>> int((0, 1))` # TypeError

**Pull request closed without merging**